data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/HistogramAggregateAction.java

```diff
@@ -180,9 +194,13 @@ public AggregateActionResponse handleEvent(final Event event, final AggregateAct
                     maxValue = doubleValue;
                 }
             }
-        } 
-        // Keep over-writing endTime to get the last time a record of this group received
-        groupState.put(endTimeKey, Instant.now());
+            Instant groupStartTime = (Instant)groupState.get(startTimeKey);
+            Instant groupEndTime = (Instant)groupState.get(endTimeKey);
+            if (eventStartTime.isBefore(groupStartTime))
```

Comment:
Are these conditions being tested? Both true and false?

Reply:
Yes because the test cases assign the times with random delta.
```
Instant sTime = (i == 0) ? testTime : testTime.plusSeconds(random.nextInt(5));                                                   
            Instant eTime = (i == testCount-1) ? testTime.plusSeconds(100) : testTime.plusSeconds (50+random.nextInt(45));  
```